scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/DTLSSession.java

```diff
@@ -418,11 +415,13 @@ public int getWriteEpoch() {
 		return writeEpoch;
 	}
 
+	// tests only, currently not used
 	void setWriteEpoch(int epoch) {
 		if (epoch < 0) {
 			throw new IllegalArgumentException("Write epoch must not be negative");
 		} else {
 			this.writeEpoch = epoch;
+			this.writeEpochTag = Integer.toString(epoch);
```

Comment:
You create an attribute to avoid to create a string each time we create a context ? I'm not sure to see the benefit.
And if there is one, should we rather store the context itself ?

Reply:
> And if there is one, should we rather store the context itself ?

that was something, I tried to introduce last year :-).

Reply:
Let me guess: I said I didn't see the benefits ? (or you didn't open the PR?)

Reply:
I can also got for convert the ints "on demand" always into strings, if you prefer.

Reply:
meaning removing `writeepochTag` and `readEpochTag` attributes ?

Reply:
yes.

Reply:
I would prefer. And if we think an optimization like this is worth it. We could go later in that way but targeting  `DtlsEndpointContext` instead.